### PR TITLE
Fix for edge case for point at poles

### PR DESCRIPTION
**Pull request by hhakimi53**
Fix edge case in canonicalize/normalize logic to not evaluate/update point for antimeridian condition when the point is at pole.

### CODE REVIEW

**Comment by wolffcm** [comment]
There is equivalent code in the GeographyPoint equals method.  Could you check to see if we have the same issue there too?  And if so, could you apply the same fix, and add a test?

File: src/ee/common/GeographyPointValue.hpp [hunk 1]
```diff
@@ -229,9 +229,9 @@ class GeographyPointValue {
             newLng = 0.0;
         }
 
-        // For longitudes within epsilon of the antimeridian
-        // (on the east side), canonicalize to 180.0.
-        if (180.0 + m_longitude < epsilon()) {
+        // If point is not at the poles, evaluate longitudes within epsilon
```
Comment (wolffcm) [inline src/ee/common/GeographyPointValue.hpp]: I just have a style comment, that I find it a little confusing to have a newline and a couple line of comments between an `if` block and its following `else` block.  This is a matter of personal style though.

**Comment by wolffcm** [comment]
That's all my comments.

File: src/ee/common/GeographyPointValue.hpp (hunk 1 above)
Comment (hhakimi53) [inline src/ee/common/GeographyPointValue.hpp]: I will move the comment inside the else condition block.

**Comment by wolffcm** [comment]
Sorry I meant to say that the Java class GeographyPoint might have the same problem, in case my earlier comment wasn't clear.

**Comment by hhakimi53** [comment]
GeographyPointValue.equals() is good. I have added test cases to verify the same.

File: src/ee/common/GeographyPointValue.hpp [hunk 2]
```diff
@@ -229,9 +229,9 @@ class GeographyPointValue {
...
+        // of the antimeridian (on the east side), canonicalize to 180.0.
```
Comment (BillWhite) [inline src/ee/common/GeographyPointValue.hpp]: I guess this works.

**Comment by BillWhite** [comment]
This seems ok to me.  I would have imagined that you would be changing the equals method, and the constructor doesn't matter that much. But maybe I don't understand it.

That's my two shekels.

**Comment by hhakimi53** [comment]
Bill, sorry, I did not get the point about the changes needed to equals method. The issue seems to be   canonicalize() that is used by compareWith() to evaluate if two geo points are equal. In canonicalize() we were losing the information whether point was at pole or not for the case the point was at antimeridian longitude. May be I am missing the point you are trying to make, please let me know. Also GeographyPoint (frontend/java) equals method works when it is asked to evaluate if two points are equal or not. So did not see need for changes in equals().

**Comment by hhakimi53** [comment]
Talked to Bill and clarified about it. Will merge the changes to master.

**Pull request merged**